packages/material-ui/src/Chip/Chip.js

```diff
@@ -371,7 +371,7 @@ class Chip extends React.Component {
 
     return (
       <Component
-        role="button"
+        role={clickable ? 'button' : 'presentation'}
```

Comment:
I don't think [`presentation`](https://www.w3.org/WAI/PF/aria/roles#presentation) is the right choice here since it basically removes any meaning from the element. I would rather set it to `undefined` so that the implicit role of the given `Component` doesn't get removed.

If we remove the `button` role we should also remove the tabIndex. Doesn't sound right to make an element focusable if it can't be interacted with.

Reply:
This makes sense, I updated solution

Reply:
Ok this new changes break tests, let me fix that as well

Reply:
`tabIndex` management was a bit weird in this component, unless I'm missing something. I removed the manual overriding, so it is now done, like `role`, within the `{...other}` props.

There is a `clickable` state based on `onClick` AND a `clickable` prop, but also a `onDelete` prop that makes the Chip interactive as well. Usage of `role` and `tabIndex` should now be correct for all use cases.

Tests pass. I removed test when `tabIndex` is not supposed to be set (non interactive element), and added a test to verify that `presentation` can be overridden  in addition to `tabIndex`. 

I also looked manually at the UX, and my local copy of documentation behaves the same as website.  (mouse and keyboard interactions)

Reply:
@mbrookes Has already been spending quite some time on the accessibility part of the component. I would halt any work until we get his feedback.